Lib/wsgiref/handlers.py

```diff
@@ -305,9 +320,6 @@ def sendfile(self):
     def finish_content(self):
         """Ensure headers and content have both been sent"""
         if not self.headers_sent:
-            # Only zero Content-Length if not set by the application (so
-            # that HEAD requests can be satisfied properly, see #3839)
-            self.headers.setdefault('Content-Length', "0")
```

Comment:
I tried that first but I think I forgot to drop the `if blocks == 1` check in `set_content_length`. I think setting `self.result = []` should work and it would even help to simplify `set_content_length`.

Reply:
Done.